tools/src/main/java/org/apache/kafka/trogdor/workload/SustainedConnectionWorker.java

```diff
@@ -0,0 +1,563 @@
+
+        // These variables are used to maintain the scheduling of the connection keep-alive.
+        protected long nextUpdate = 0;
+        protected boolean inUse = false;
+
+        @Override
+        public boolean needsRefresh() {
+            return !this.inUse && (System.currentTimeMillis() > this.nextUpdate);
+        }
+
+        @Override
+        public void claim() {
+            this.inUse = true;
+        }
+
+        @Override
+        public void close() throws Exception {
+            this.closeQuietly();
+        }
+
+        protected abstract void closeQuietly();
+
+    }
+
+    private class MetadataSustainedConnection extends ClaimableConnection {
+
+        // This variable is used to maintain the connection itself.
+        private AdminClient client;
+
+        // This variable is used to maintain the scheduling of the connection keep-alive.
+        private long refreshRate;
+
+        // This variable is used to maintain the connection properties.
+        private final Properties props;
+
+        MetadataSustainedConnection() {
+            // This variable is used to maintain the connection itself.
+            this.client = null;
+
+            // This variable is used to maintain the scheduling of the connection keep-alive.
+            this.refreshRate = SustainedConnectionWorker.this.spec.refreshRateMs();
+
+            // This variable is used to maintain the connection properties.
+            this.props = new Properties();
+            this.props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, SustainedConnectionWorker.this.spec.bootstrapServers());
+            WorkerUtils.addConfigsToProperties(
+                    this.props, SustainedConnectionWorker.this.spec.commonClientConf(), SustainedConnectionWorker.this.spec.commonClientConf());
+        }
+
+        @Override
+        public void refresh() {
+            try {
+                if (this.client == null) {
+                    // Housekeeping to track the number of opened connections.
+                    SustainedConnectionWorker.this.totalMetadataConnections.incrementAndGet();
+
+                    // Create the admin client connection.
+                    this.client = AdminClient.create(this.props);
+                }
+
+                // Fetch some metadata to keep the connection alive.
+                this.client.describeCluster().nodes().get();
+
+            } catch (Throwable e) {
+                // Set the admin client to be recreated on the next cycle.
+                this.closeQuietly();
+
+                // Housekeeping to track the number of opened connections and failed connection attempts.
+                SustainedConnectionWorker.this.totalMetadataConnections.decrementAndGet();
+                SustainedConnectionWorker.this.totalMetadataFailedConnections.incrementAndGet();
+                SustainedConnectionWorker.log.error("Error while refreshing sustained AdminClient connection", e.getMessage());
```

Comment:
This won't log the error. We want to use the `public void error(String msg, Throwable t);` overload, so we need to change `e.getMessage()` -> `e`

Reply:
Done.